catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -300,22 +300,34 @@ public SolrQuery propertyIsLike(String propertyName, String pattern, boolean isC
 
     String searchPhrase = escapeSpecialCharacters(pattern);
 
-    if (Metacard.ANY_TEXT.equals(propertyName) && SOLR_WILDCARD_CHAR.equals(searchPhrase)) {
+    boolean isAnyText = Metacard.ANY_TEXT.equals(propertyName);
+
+    if (isAnyText && SOLR_WILDCARD_CHAR.equals(searchPhrase)) {
       return new SolrQuery("*:*");
     }
 
-    if (searchPhrase.contains(SOLR_WILDCARD_CHAR)
-        || searchPhrase.contains(SOLR_SINGLE_WILDCARD_CHAR)) {
+    boolean isWildcard =
+        searchPhrase.contains(SOLR_WILDCARD_CHAR)
+            || searchPhrase.contains(SOLR_SINGLE_WILDCARD_CHAR);
+
+    boolean isQuotedPhrase = pattern.startsWith("\"") && pattern.endsWith("\"");
```

Comment:
So this only works if the whole thing is quoted.  What if I wanted to do a search where part of the search phrase was quoted?  For example: `"abc 123" def "ef gh"`.  Couldn't we grab the quoted sections and escape the whitespace characters within the quotes?  Also, my example creates an interesting problem for this code because it actually does start and end with a quote

Reply:
I did not want to introduce a large, complex grammar to LIKE values.  I figured for those other use cases we can use the existing logical operators like AND and OR already available in filters.

I went back and forth on if I should detect that kind of example by making sure there were only two double quotes.  I am hoping we can hide this detail from users and instead provide ways through the UI to create this type of query.  So ideally a user would never have to quote their own values.

Reply:
Might also be possible to do a transform on the backend as a `PreQuery` plugin. I could see your query, `"abc 123" def "ef gh"`, getting decomposed to something like `"abc 123" def*` **AND** `*def "ef gh"` if we wanted the Solr provider to meet you halfway, which still isn't perfect but might be good enough. 

Even less perfect would be a representation like `"abc 123"*` **AND** `*def*` **AND** `*"ef gh"` but it seems either way the Solr side needs to provide a little bit more flexibility.

Reply:
I would like to avoid adding the complexity of grammar parsing when we can get the same query from building it up at the filter level already.

Reply:
I agree. I don't think you need full grammar parsing at the Solr level. 

But I also don't think you can fully decompose "exact" queries and get expected results because breaking a LIKE predicate apart at the logic level would involve adding wildcard symbols to reassemble semantic equivalence to the original LIKE pattern. And if you only detect starting with quotes and ending with quotes, then something like `*text "literal"` won't get interpreted correctly. 

Could the `wildcardSolrQuery` function get support for partial "exactness"?

Reply:
Thinking about this more I might have incorrectly assumed that order matters for a `PropertyIsLike` predicate. I suspect @derekwilhelm's original query `"abc 123" def "ef gh"` could actually be `"abc 123"` **AND** `def` **AND** `"ef gh"` in which case the Solr layer would work as is. 

But if term order matters within a `PropertyIsLike`, then these kinds of query transforms wouldn't work. Let me know if my reasoning is not correct.

Reply:
I'm fine leaving it the way that it is for now.  I was just wondering if there was an easy way to expand the capability to cover more cases.  We don't really handle the scenario I presented at a higher level unless someone specifically constructs an advanced search that way.  With the growing desire to do more direct logic within a text search this seemed like a good place to account for more cases.  However, it may be more appropriate to do that somewhere else.